#Imports

In [2]:
import pandas as pd
import numpy as np

#Questão 1
Qual o número de caso de seguro - 1338

Quantos atributos são numéricos - 4

Há quantos valores diferentes para região - 4

Qual a maior quantidade de filhos - 5

In [22]:
insurence = pd.read_csv('https://raw.githubusercontent.com/Rogerio-mack/IMT_Ciencia_de_Dados/main/data/insurance.csv')
insurence.head()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [23]:
insurence.describe()

age          bmi     children       charges
count  1338.000000  1338.000000  1338.000000   1338.000000
mean     39.207025    30.663397     1.094918  13270.422265
std      14.049960     6.098187     1.205493  12110.011237
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.296250     0.000000   4740.287150
50%      39.000000    30.400000     1.000000   9382.033000
75%      51.000000    34.693750     2.000000  16639.912515
max      64.000000    53.130000     5.000000  63770.428010

#Questão 2

Qual a quantidade de casos por região?

In [21]:
insurence['region'].value_counts()

southeast    364
southwest    325
northwest    325
northeast    324
Name: region, dtype: int64

#Questão 3

Qual a média dos pagamentos (charges) efetuados - 13270.422..

#Questão 4

Qual a média dos pagamentos (charges) dos 10 maiores pagamentos efetuados - 49024

In [37]:
insurence_top10 = insurence.sort_values('bmi',ascending = False).head(10)
insurence_top10.describe()

age        bmi   children       charges
count  10.000000  10.000000  10.000000     10.000000
mean   40.900000  49.024000   1.300000  20922.647071
std    14.873167   2.260016   1.494434  21961.333624
min    18.000000  46.750000   0.000000   1163.462700
25%    26.500000  47.540000   0.250000   8421.171175
50%    46.500000  47.905000   1.000000  10565.118000
75%    52.000000  50.050000   1.750000  36524.182275
max    58.000000  53.130000   5.000000  63770.428010

#Questao 5

Calcule a distância interquartis dos pagamentos.
Dica Empregue a função quantile() do Pandas e use IQR = Q3 - Q1.